db/tables.sql

```diff
@@ -0,0 +1,22 @@
+CREATE TABLE public.streams
+(
+    id              bigserial         NOT NULL,
+    context         character varying NOT NULL,
+    stream_name     character varying NOT NULL,
+    stream_id       character varying NOT NULL,
+    stream_revision int DEFAULT -1    NOT NULL
+);
+
+CREATE TABLE public.events
+(
+    id              uuid                        NOT NULL DEFAULT public.gen_random_uuid(),
+    stream_id       bigint                      NOT NULL,
+    type            character varying,
```

Comment:
Links are records with not null `link_id` column. We already can recognize them. But may be you are right, because user may want to read links only by supplying `event_types: ['$>']` filter. Will improve this.